libraries/render-utils/src/MeshPartPayload.cpp

```diff
@@ -493,8 +493,7 @@ void ModelMeshPartPayload::render(RenderArgs* args) {
         batch.setDrawcallUniform(drawcallInfo);
     }
 
-    if (!_drawMaterials.empty() && _drawMaterials.top().material && _drawMaterials.top().material->isProcedural() &&
-            _drawMaterials.top().material->isReady()) {
+    if (_shapeKey.hasOwnPipeline()) {
```

Comment:
This is the fix.  Before, it was possible for a ModelMeshPartPayload to have a non-custom shape key even though it was supposed to render with a shader.  In that case, it would render in the middle of when all the other models rendered, but would switch the active program to the custom shader, so all models after it would render with the custom shader.  Now, we use the shape key as the voice of truth, and make sure to update the shape key as necessary in RenderableModelEntityItem.cpp